File: PeptideHitResultsProcessor/clsSearchResultsMODPlus.vb
Option Strict On

' This class is used to track the peptide details for a MODPlus search result
' See clsSearchResultsBaseClass for additional information
'
' -------------------------------------------------------------------------------
' Written by Matthew Monroe for the Department of Energy (PNNL, Richland, WA)
' Created 5/15/2015
'
' E-mail: matthew.monroe@pnnl.gov or matt@alchemistmatt.com
' Website: http://omics.pnl.gov/ or http://www.sysbio.org/resources/staff/
' -------------------------------------------------------------------------------

Imports PHRPReader

Public Class clsSearchResultsMODPlus
    Inherits clsSearchResultsBaseClass

#Region "Classwide Variables"
    ' Note that "Automatic properties" are being used; thus, we don't need to explicitly define class variables 
#End Region

#Region "Properties"

    ' Auto-Properties	
    Public Property Spectrum_Index As String

    Public Property Precursor_mz As String              ' Observed precursor m/z, converted to monoisotopic mass by MODa

    Public Property MODPlusComputedDelM As String
    Public Property MODPlusComputedDelMPPM As String

    Public Property MODPlusScore As String

    Public Property Probability As String

    Public Property PeptidePosition As String

    Public Property ModificationAnnotation As String


#End Region

    Public Sub New(objPeptideMods As clsPeptideModificationContainer, peptideSeqMassCalculator As clsPeptideMassCalculator)
        ' Note that the following call will call both the base class's Clear sub and this class's Clear Sub
        MyBase.New(objPeptideMods, peptideSeqMassCalculator)
    End Sub

    Public Overrides Sub Clear()
        MyBase.Clear()
        Spectrum_Index = String.Empty

        Precursor_mz = String.Empty

        MODPlusComputedDelM = String.Empty
        MODPlusComputedDelMPPM = String.Empty

        MODPlusScore = String.Empty

        Probability = String.Empty

        PeptidePosition = String.Empty
    End Sub

End Class

File: PeptideHitResultsProcessor/clsSearchResultsMODPlus.cs
// -------------------------------------------------------------------------------
// Written by Matthew Monroe for the Department of Energy (PNNL, Richland, WA)
//
// E-mail: matthew.monroe@pnnl.gov or proteomics@pnnl.gov
// Website: https://omics.pnl.gov/ or https://www.pnnl.gov/sysbio/ or https://panomics.pnnl.gov/
// -------------------------------------------------------------------------------
//
// Licensed under the 2-Clause BSD License; you may not use this file except
// in compliance with the License.  You may obtain a copy of the License at
// https://opensource.org/licenses/BSD-2-Clause
//
// Copyright 2018 Battelle Memorial Institute

using PHRPReader;

namespace PeptideHitResultsProcessor
{
    /// <summary>
    /// This class is used to track the peptide details for a MODPlus search result
    /// See clsSearchResultsBaseClass for additional information
    /// </summary>
    public class clsSearchResultsMODPlus : clsSearchResultsBaseClass
    {
        #region "Classwide Variables"
        // Note that "Automatic properties" are being used; thus, we don't need to explicitly define class variables
        #endregion

        #region "Properties"

        // Auto-Properties
        public string Spectrum_Index { get; set; }

        public string Precursor_mz { get; set; }            // Observed precursor m/z, converted to monoisotopic mass by MODa

        public string MODPlusComputedDelM { get; set; }
        public string MODPlusComputedDelMPPM { get; set; }

        public string MODPlusScore { get; set; }

        public string Probability { get; set; }

        public string PeptidePosition { get; set; }

        public string ModificationAnnotation { get; set; }

        #endregion

        // Note that the following call will call both the base class's Clear method and this class's Clear method
        public clsSearchResultsMODPlus(clsPeptideModificationContainer peptideMods, clsPeptideMassCalculator peptideSeqMassCalculator)
            : base(peptideMods, peptideSeqMassCalculator)
        {
        }

        public override void Clear()
        {
            base.Clear();
            Spectrum_Index = string.Empty;

            Precursor_mz = string.Empty;

            MODPlusComputedDelM = string.Empty;
            MODPlusComputedDelMPPM = string.Empty;

            MODPlusScore = string.Empty;

            Probability = string.Empty;

            PeptidePosition = string.Empty;
        }
    }
}
